### ISSUE TITLE
Jadeite version check

### ISSUE THREAD

**Issue by dalehenrich** [ericwinger/Jade#521]
Customer has requested that we institute a jadeite version check so that users won't be caught be surprise when using an outdated Jadeite ... I've recently used an older version of Jadeite (out of laziness) and discovered that the Jadeite allowed me to start using the project browser and do a little bit of work/browsing before hitting an error "that would not go away" (could not exit session without closing all of the open windows) ... 

I would think that it would be sufficient to have pop up a dialog indicating that it is an invalid version and allowing the user to go ahead anyway or terminate the session immediately ...

This is not needed for the November release, but would be nice, so I've assigned it November label, but it can be dropped ...

**Comment by LisaAlmarode** [comment]
Is this a check of Jadeite version vs. Rowan loaded version?  If I can't login with an unmatching jadeite, I won't be able to upgrade from inside Jadeite, right?  I don't know what the official process is but I would like to do everything from jadeite, I just don't have the ability to checkout on a tag.

**Comment by dalehenrich** [comment]
@ericwinger and I talked about this bit and I was thinking that a warning dialog would pop up if you connected to an image with a Jadeite client that didn't "match" the Jadeite service method version ... my observation is that when you use a mismatched client you can get real work done for a period of time and then the system can lock up and your only alternative is to quit ... which would be a real pain if you lost meaningful work ...

**Comment by ericwinger** [comment]
Some thoughts:
- I think we'll need a "two-way" check. One from client -> server. The other from server -> client. Otherwise, there won't be a way to validate clients or servers that predated the check.  I'm not sure if there's a pre-existing server hook to catch an older client. Alternatively, we don't worry about that and just do a client-initiated check going forward. The latter is easier. 
- Currently, the Jadeite version is tied to a git tag. We would need to make the server become aware of it's git tag. That tag could also just be a method constant that's manually updated during release after the Jadeite client is built. (Alternatively, we could use an internal consistency check like a hard-coded constant in a client & server method. That would be easier to code, but opaque to the user which is not desirable)
- Longstanding bug #147 talks about upgrading services & JadeServer packages so Jadeite can be upgraded without upgrading Rowan. I presume that work won't be done at this time. 
- Check should be done only once after login. No visual indication is given if versions match. 
- A mismatch would not prevent login, only raise a warning dialog.  I suppose we could also turn the login window red or some other background gui warning to serve as a constant passive warning.

**Comment by ericwinger** [comment]
Have a version check system implemented. Fairly simplistic.  Checks the server & client versions which are held in class side methods in ``RowanService>>version``. If a client doesn't find that method or the server doesn't get sent that message, logins fail. If the versions don't match, logins fail. Warnings are provided with version numbers it was comparing. Version numbers are integers that represent the current Jadeite version.

**Comment by LisaAlmarode** [comment]
PLEASE don't fail... there should be a warning, yes, but logins should not fail; this was the agree-on behavior.

I log in with a mismatch version so I can then load the latest Rowan code...

**Comment by ericwinger** [comment]
@LisaAlmarode - Duly noted. No logout is done on a version failure.

**Comment by LisaAlmarode** [comment]
If you take the new Jadeite 8.0.82 and try to login to a repository with Oscar 3.0.81, it puts up a not very appropriate error and does not allow login.  This isn't a regression per se since this didn't always work before, but it would be nice to fails in a slightly more informative way...

![image](https://user-images.githubusercontent.com/13142166/67350838-d2fb4b00-f500-11e9-9f01-6d69676c8624.png)

**Comment by LisaAlmarode** [comment]
I will note that when I was using Jadeite 3.0.81 with Oscar-3.0.81, and loaded Oscar-3.0.82, it did give me an informative warning; and in Jadeite 3.0.81 on login to Oscar 3.0.82 is gives me an informative warning.

**Comment by ericwinger** [comment]
Should be fixed in 3.0.83.

**Comment by LisaAlmarode** [comment]
It's not a nice error if you try to login 3.0.58 or 3.0.67- it respectively gets a walkback and an error dialog. But these are old enough where elegance may be impossible.
